src/main/kotlin/com/anaplan/engineering/vdmgradleplugin/TypeChecker.kt

```diff
@@ -64,28 +65,28 @@ open class VdmTypeCheckTask(private val includeTests: Boolean) : OvertureTask()
         if (specificationFiles.isEmpty) {
             logger.info("No files found")
         } else {
-            standardOutput = ByteArrayOutputStream()
+            clearLogFile()
+            standardOutput = LogAndSaveOutputStream(logFile) { log(it) }
             jvmArgs = project.vdmConfig.overtureJvmArgs
             setArgs(constructArgs())
             classpath = project.files(createClassPathJar())
             super.exec()
-            writeOutput(standardOutput)
         }
     }
 
-    private fun writeOutput(os: OutputStream) {
-        val output = os.toString()
-        logFile.writeText(output)
-        output.lineSequence().forEach {
-            logger.log(getLogLevel(it), it)
+    private fun log(line: String) {
+        val level = when {
+            line.startsWith("Error") -> LogLevel.ERROR
+            line.startsWith("Warning") -> LogLevel.WARN
+            line.startsWith("Parsed") || line.startsWith("Type checked") -> LogLevel.INFO
+            else -> LogLevel.LIFECYCLE
         }
+        logger.log(level, line)
     }
 
-    private fun getLogLevel(line: String) = when {
-        line.startsWith("Error") -> LogLevel.ERROR
-        line.startsWith("Warning") -> LogLevel.WARN
-        line.startsWith("Parsed") || line.startsWith("Type checked") -> LogLevel.INFO
-        else -> LogLevel.LIFECYCLE
+    private fun clearLogFile() {
```

Comment:
is this necessary?

if you're not telling the stream to append doesn't it just overwrite any existing file anyway?

Reply:
Indeed, I am appending to the stream: https://github.com/anaplan-engineering/vdm-gradle-plugin/blob/38388132a9b52d388b62fc36170a9d61e80326a5/src/main/kotlin/com/anaplan/engineering/vdmgradleplugin/TypeChecker.kt#L106